lib/cli/drive-selection-widget.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const chalk = require('chalk');
+const inquirer = require('inquirer');
+const _ = require('lodash');
+
+const DynamicList = require('./dynamic-list');
+const scanner = require('./drive-scanner');
+const driveComparer = require('../shared/utils').driveComparer;
+const constraints = require('../shared/drive-constraints');
+
+inquirer.registerPrompt('dynamicList', DynamicList);
+
+const driveToChoiceFn = (image) => {
+
+  /**
+   * Map a drive object to a `inquirer` choice object
+   * @param {object} drive - Drive object as returned by `drivelist`
+   * @private
+   * @returns {{name: string, value: object}}
+   */
+  const driveToChoice = (drive) => {
+    const size = drive.size / 1000000000;
+
+    const driveInfo = `${drive.device} (${size.toFixed(1)} GB) - ${drive.description}`;
```

Comment:
Actually, this WHOLE logic would be re-used between the CLI and the GUI, and not just the byte to gigabyte function I mentioned above. The GUI displays the exact same thing, but it creates separate HTML elements for the labels.

We can build a generic function that returns an object that looks like this:

```javascript
{
   description: '/dev/disk2 (2 GB) - Foo bar',
   labels: [
      'SYSTEM DRIVE',
      'LOCKED'
   ]
}
```

Then, in the GUI, we can loop between the labels in Angular and create elements, while in the CLI, we can join the label values and append it to the description.